src/polls/PollForm.vue

```diff
@@ -199,9 +292,50 @@ export default defineComponent({
     ChangeDate(value: Date) {
       this.pollData.creationDate = value;
     },
+    onChangeCreationDate() {
+      const dateTomorrow = new Date();
+      dateTomorrow.setDate(dateTomorrow.getDate() + 1);
+      dateTomorrow.setHours(0, 0, 0, 0);
+      if (this.pollData.creationDateInEdition < dateTomorrow) {
+        this.isDisabled = true;
+        this.isCreationDateHelpActive = true;
+      } else {
+        this.beginDateInEdition.setHours(0, 0, 0, 0);
+
+        this.finishDate = new Date(this.pollData.creationDateInEdition.getTime() + 14 * 86400000);
+        this.finishDate.setHours(23, 59, 59, 59);
+        this.pollData.completionDate = this.finishDate.toLocaleString('uk-UA').split(',')[0];
+
+        this.isCreationDateHelpActive = false;
+        this.isDisabled = false;
+      }
+    },
+    async initData() {
+      await this.$store.dispatch(
+        `${StoreModuleEnum.pollsStore}/${PollsActionEnum.SET_SELECTED_POLL}`,
+        this.$props.poll.id
+      );
+
+      this.pollData.title = this.selectedPoll?.header ?? '';
+      this.pollData.description = this.selectedPoll?.description ?? 'Повний опис опитування';
+      this.pollData.polledHouses = this.selectedPoll?.polledHouses ?? [];
+      this.pollData.creationDateInEdition = this.selectedPoll?.creationDate.toLocaleString('uk-UA');
+      this.pollData.completionDate = this.selectedPoll?.completionDate.toLocaleString('uk-UA');
```

Comment:
Why do you need this default values set to the poll data if you can set them in data() function?

Reply:
yes, you're right, totally agree with your suggestion

Reply:
@aleonovdementyev, I was wondering to know if I may not put 'Повний опис опитування' as dafault in data()?
There are some issues:
1) technical: ternary operator like this !this.isEditing ? '' : 'Повний опис опитування' (and some other its versions) does not work in option of description in data(). I also attempted to pass via computed expression (for example), but it was failed.
2) logical: according to this default meaning of description, if there is no desc in polledData, please show 'Повний опис опитування'. But it is impossible to be empty because the poll form can't be sumbitted in this case, it becomes invalid.
Also, it is related to all other default values put in lines 319-321 — any data must be written there.
Here is another problem: we do not receive data from polledData, therefore I just suppose that previous engineer put default values to avoid errors.
To sum up, I would suggest not mentioning 'Повний опис опитування' as default in data() as we are passing in all other parameters initial (=empty) values.


Also, @yevhendevbox and @anatoliy-stepanchenko, if you have any other ideas how to put conditional expression in value in data(), I will be so thankful for your suggestions!